File: Notifications/Form1.cs
using FluentScheduler;
using Microsoft.Toolkit.Uwp.Notifications;
using Notifications.Classes;
using ToastLibrary;
using ToastLibrary.Models;
using static Notifications.Classes.Dialogs;

namespace Notifications;

public partial class Form1 : Form
{
    public Form1()
    {
        InitializeComponent();

        if (!ToastOperations.ListenerAvailable())
        {
            MessageBox.Show(@"OS does not support code in this app");
            Controls.OfType<Button>().ToList().ForEach(button =>
            {
                button.Enabled = false;
            });
        }

        ToastOperations.OnInterceptHandler += ToastOperations_OnInterceptHandler;
    }

    /// <summary>
    /// Update intercept button text, must check control.InvokeRequired as the code
    /// is executed from another thread.
    /// </summary>
    private void ToastOperations_OnInterceptHandler(int sender)
    {
        InterceptButton.InvokeIfRequired(b => b.Text = $@"Intercept count [{sender}]");
    }
    private void HeroButton_Click(object sender, EventArgs e)
    {
        ToastOperations.Hero();
    }

    private void ScheduleButton_Click(object sender, EventArgs e)
    {
        ToastOperations.Schedule();
    }

    private void AlarmButton_Click(object sender, EventArgs e)
    {
        ToastOperations.Alarm();
    }

    private void TextInputButton_Click(object sender, EventArgs e)
    {
        ToastOperations.TextBoxFavoriteColor();
    }

    private void SelectButton_Click(object sender, EventArgs e)
    {
        ToastOperations.ComboBoxFavoriteColor();
    }

    private async void DownLoadButton_Click(object sender, EventArgs e)
    {
        List<PublicHoliday> usHolidays = await WorkOperations.GetHolidays();
        if (usHolidays.Any())
        {
            ToastOperations.HolidaysDownloaded();
        }
        else
        {
            ToastOperations.HolidaysDownloadFailed();
        }
    }

    private async void InterceptButton_Click(object sender, EventArgs e)
    {
        InterceptButton.Enabled = false;
        await Helpers.SimulateWorkAsync();
        InterceptButton.Enabled = true;

        var karenPhoto = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Images", @"Karen.png");

        new ToastContentBuilder()
            .AddArgument("action", "viewConversation")
            .AddArgument("conversationId", ToastOperations.Dictionary["key2"])
            .AddText("Hey")
            .SetToastDuration(ToastDuration.Short)
            .AddAppLogoOverride(new Uri(karenPhoto), ToastGenericAppLogoCrop.Circle)
            .AddButton(new ToastButton().SetContent("Go for it")
                .AddArgument("action", "viewReport")
            )
            .Show(toast =>
            {
                toast.ExpirationTime = DateTime.Now.AddMinutes(2);
            });
    }

    private void StopScheduledNotificationButton_Click(object sender, EventArgs e)
    {
        Schedule? schedule = JobManager.AllSchedules.FirstOrDefault(x => x.Name == "Annoying");
        if (schedule is not null)
        {
            JobManager.RemoveJob("Annoying");
        }
    }

    private void SelectionButton_Click(object sender, EventArgs e)
    {
        ToastOperations.SelectionBox();
    }

    private async void TaxpayerButton_Click(object sender, EventArgs e)
    {
        if (await DataOperations.DatabaseExist())
        {
            if (Question("Truncate Taxpayer table", "Question"))
            {
                DataOperations.TruncateTable();
            }

            var (success, exception) = await DataOperations.AddNewTaxpayers(BogusOperations.Taxpayers());
            if (exception is not null)
            {
                // no toast here, we want this in the user's face
                MessageBox.Show($@"Operation failed {exception.Message}");
            }
            else
            {
                ToastOperations.FinishedAddingRecords();
            }
        }
        else
        {
            MessageBox.Show("Database not found");
        }

    }


}

File: NotificationsVisualBasic/Form1.vb
Imports System.IO
Imports Microsoft.Toolkit.Uwp.Notifications

Public Class Form1
    Private Sub AlarmButton_Click(sender As Object, e As EventArgs) Handles AlarmButton.Click
        Alarm()
    End Sub

    Public Shared Sub Alarm()

        Dim alarmPhoto = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Images", "alarm.png")
        Dim checkPhoto = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Images", "checkMark.png")

        Call (New ToastContentBuilder()).AddArgument("action", "viewConversation").
            AddArgument("conversationId", "alarmKey").
            AddText("Time for work").
            AddButton(
                (New ToastButton()).SetContent("OK").AddArgument("action", "work").
                         SetImageUri(New Uri(checkPhoto))).AddButton(
                             (New ToastButton()).SetContent("Snooze").
                                                                        AddArgument("action", "snooze").
                                                                        SetImageUri(New Uri(alarmPhoto))).
            SetToastScenario(ToastScenario.Alarm).Show()
    End Sub

    Public Sub OnActivatedToast()
        AddHandler ToastNotificationManagerCompat.OnActivated,
            Sub(toastArgs)
                Dim args As ToastArguments = ToastArguments.Parse(toastArgs.Argument)
                If args("conversationId") = "alarmKey" Then
                    If args("action") = "snooze" Then
                        WorkOperations.Snooze()
                        Label1.InvokeIfRequired(
                            Sub(item)
                                item.Text = "Snooze"
                            End Sub)
                    ElseIf args("action") = "work" Then
                        WorkOperations.GotoWork()
                        Label1.InvokeIfRequired(
                            Sub(item)
                                item.Text = "Go to work"
                            End Sub)
                    End If

                End If
            End Sub
    End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        OnActivatedToast()
    End Sub
End Class